### PR TITLE
Add initial map generation

### LINKED ISSUE #28: Implement Hardcoded Map Generation
For structural reasons, it is prudent to implement a hardcoded, close to useless version of map generation so that the rest of the team can interface with it.

See #1 for the full map generation.

### PR DESCRIPTION
**Pull request by 6167656e74323431**
Adds a hard coded version of map generation so I can start to work on my other use cases (and so can anyone else that depends on an overworld existing).

Makes progress towards #1

Fixes #28

### CODE REVIEW

**Review by sbrad044 — APPROVED** [review]
Very neatly and clearly explained and written, nice

**Review by siddharthgowda — APPROVED** [review]

**Review by zhaoan12 — APPROVED** [review]
Nice class. Like you say in the docstring, we can make it adhere to clean architecture later.

**Review by hoodiexxx — COMMENTED** [review]
good implementation for MapGenerator and that is pretty useful when the ReplayGenerator collaborate with it.

File: src/main/java/com/mg105/use_cases/MapGenerator.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     * Creates a new MapGenerator that acts on a GameState.
+     *
+     * state must not be null.
+     *
+     * @param state the game state.
+     */
+    public MapGenerator(GameState state) {
+        this.state = state;
+    }
+
```
Comment (hoodiexxx) [inline src/main/java/com/mg105/use_cases/MapGenerator.java:22]: no sure whether the MapGenerator class needs a variable that contains GameState as an instance variable, but still good coding.

**Pull request merged**